plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeTransformUtils.kt

```diff
@@ -168,15 +168,13 @@ suspend fun JobId.pollTransformationStatusAndPlan(
 
     try {
         waitUntil(
-            succeedOn = { state in succeedOn },
-            failOn = { state in failOn },
+            succeedOn = { result -> result in succeedOn },
+            failOn = { result -> result in failOn },
             maxDuration = maxDuration,
             exceptionsToStopOn = setOf(
                 InternalServerException::class,
                 ValidationException::class,
-                AccessDeniedException::class,
```

Comment:
Second Davids comment?

Can we catch the `AccessDeniedException` and `SdkClientException` and then do a subsequent check to make sure we want to continue or cancel? Any metadata fields from the exceptions to look at?

This may require a more in depth review.

Reply:
We will only throw AccessDeniedException if we retry refreshing token 10 times. It shouldn't be part of exceptionsToStopOn because it will cause waitUntil to exit without actually throwing the exception.

I fixed the type and added another catch block outside, so that AccessDeniedException will return a failed PollingResult.